yt_dlp/extractor/iprima.py

```diff
@@ -134,10 +134,35 @@ def _real_extract(self, url):
         ), webpage, 'real id', group='id', default=None)
 
         if not video_id:
-            nuxt_data = self._search_nuxt_data(webpage, video_id, traverse='data')
+            nuxt_data = self._search_nuxt_data(webpage, video_id, traverse='data', fatal=False)
+
             video_id = traverse_obj(
                 nuxt_data, (..., 'content', 'additionals', 'videoPlayId', {str}), get_all=False)
 
+        if not video_id:
+            nuxt_data = self._parse_json(
+                self._search_regex(
+                    r'(?s)<script[^>]+\bid=["\']__NUXT_DATA__["\'][^>]+>(.+?)</script>',
+                    webpage, 'nuxt data'),
+                'nuxt data')
+
+            def extract_video_id_from_nuxt_data(data):
+                if not isinstance(data, list):
+                    return None
+                for itm in data:
+                    if not isinstance(itm, str):
+                        continue
+                    if len(itm) < 2:
+                        continue
+                    if itm[0] != 'p':
+                        continue
+                    if any(not ch.isdigit() for ch in itm[1:]):
+                        continue
+                    return itm
+                return None
+
+            video_id = extract_video_id_from_nuxt_data(nuxt_data)
```

Comment:
Why so complex? Just use regex

```suggestion
            video_id = traverse_obj(nuxt_data, lambda _, v: re.fullmatch(x, r'p\d+'), get_all=False)
```

Reply:
it's because I haven't really grokked how this traverse_obj magic works yet, so to me the above is 'simpler' in a way

but it's probably better to do it as you suggested in your one liner

also `re.fullmatch` args were swapped in your suggestion, have fixed that and tested it works

Reply:
In this case, `traverse_obj` is just there to make `re.fullmatch` not crash when the item is not a string. The code is otherwise equivalent to:
```py
video_id = next(x for x in next_data if re.fullmatch(r'p\d+', x))
```